test/metric_value_benchmark/metrics_value_benchmark_test.go

```diff
@@ -45,37 +44,40 @@ func init() {
 
 var (
 	ecsTestRunners []*ECSTestRunner
-	ec2TestRunners []*TestRunner
+	ec2TestRunners []*test_runner.TestRunner
 )
 
 func getEcsTestRunners(env *environment.MetaData) []*ECSTestRunner {
 	if ecsTestRunners == nil {
-		factory := &metric.MetricFetcherFactory{Env: env}
+		factory := dimension.GetDimensionFactory(*env)
 
 		ecsTestRunners = []*ECSTestRunner{
-			{testRunner: &ContainerInsightsTestRunner{ECSBaseTestRunner{MetricFetcherFactory: factory}},
-				agentRunStrategy: &ECSAgentRunStrategy{}},
+			{
+				testRunner:       &ContainerInsightsTestRunner{Base: test_runner.BaseTestRunner{DimensionFactory: factory}},
+				agentRunStrategy: &ECSAgentRunStrategy{},
+				env:              *env,
+			},
 		}
 	}
 	return ecsTestRunners
 }
 
-func getEc2TestRunners(env *environment.MetaData) []*TestRunner {
+func getEc2TestRunners(env *environment.MetaData) []*test_runner.TestRunner {
+	factory := dimension.GetDimensionFactory(*env)
```

Comment:
any reason to move this line above the if block?

Reply:
There was a reason when I first wrote something in this file but no not anymore